File: javafx-app/src/main/java/com/santea/model/Teleconsultation.java
package com.santea.model;

import java.time.LocalDateTime;

public class Teleconsultation {
    private Integer id;
    private User initiator;
    private User recipient;
    private String roomName;
    private String description;
    private LocalDateTime scheduledAt;
    private LocalDateTime startedAt;
    private LocalDateTime endedAt;
    private String status;
    private Integer durationSeconds;
    private String type;
    private LocalDateTime createdAt;

    public Teleconsultation() {
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public User getInitiator() {
        return initiator;
    }

    public void setInitiator(User initiator) {
        this.initiator = initiator;
    }

    public User getRecipient() {
        return recipient;
    }

    public void setRecipient(User recipient) {
        this.recipient = recipient;
    }

    public String getRoomName() {
        return roomName;
    }

    public void setRoomName(String roomName) {
        this.roomName = roomName;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public LocalDateTime getScheduledAt() {
        return scheduledAt;
    }

    public void setScheduledAt(LocalDateTime scheduledAt) {
        this.scheduledAt = scheduledAt;
    }

    public LocalDateTime getStartedAt() {
        return startedAt;
    }

    public void setStartedAt(LocalDateTime startedAt) {
        this.startedAt = startedAt;
    }

    public LocalDateTime getEndedAt() {
        return endedAt;
    }

    public void setEndedAt(LocalDateTime endedAt) {
        this.endedAt = endedAt;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Integer getDurationSeconds() {
        return durationSeconds;
    }

    public void setDurationSeconds(Integer durationSeconds) {
        this.durationSeconds = durationSeconds;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public String getJitsiRoomName() {
        return roomName;
    }

    public void setJitsiRoomName(String roomName) {
        this.roomName = roomName;
    }

    public String getJitsiRoomUrl() {
        return "https://meet.jit.si/" + (roomName != null ? roomName : "");
    }

    public Integer getPatientId() {
        return recipient != null ? recipient.getId() : null;
    }

    public Integer getProfessionalId() {
        return initiator != null ? initiator.getId() : null;
    }

    public Integer getDurationMinutes() {
        if (durationSeconds == null) return 0;
        return durationSeconds / 60;
    }

    public void startConsultation() {
        this.status = "ongoing";
        this.startedAt = LocalDateTime.now();
    }

    public void endConsultation() {
        this.status = "completed";
        this.endedAt = LocalDateTime.now();
        if (startedAt != null) {
            this.durationSeconds = (int) java.time.temporal.ChronoUnit.SECONDS.between(startedAt, endedAt);
        }
    }

    public void cancelConsultation() {
        this.status = "cancelled";
    }

    public boolean isActive() {
        return isOngoing();
    }

    public boolean isOngoing() {
        return "ongoing".equalsIgnoreCase(status);
    }

    public boolean isPending() {
        return "pending".equalsIgnoreCase(status);
    }

    public boolean isRequested() {
        return "requested".equalsIgnoreCase(status);
    }

    public boolean isCompleted() {
        return "completed".equalsIgnoreCase(status);
    }

    public boolean isCancel() {
        return "cancelled".equalsIgnoreCase(status);
    }

    public String getDurationFormatted() {
        if (durationSeconds == null || durationSeconds <= 0) {
            return "-";
        }

        int hours = durationSeconds / 3600;
        int minutes = (durationSeconds % 3600) / 60;
        int seconds = durationSeconds % 60;
        StringBuilder out = new StringBuilder();
        if (hours > 0) {
            out.append(hours).append("h");
        }
        if (minutes > 0) {
            if (out.length() > 0) out.append(" ");
            out.append(minutes).append("m");
        }
        if (seconds > 0 || out.length() == 0) {
            if (out.length() > 0) out.append(" ");
            out.append(seconds).append("s");
        }
        return out.toString();
    }

    public String getNotes() {
        return description;
    }

    public void setNotes(String notes) {
        this.description = notes;
    }

    public LocalDateTime getUpdatedAt() {
        return endedAt != null ? endedAt : startedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        if (endedAt == null) {
            this.endedAt = updatedAt;
        }
    }
}

File: symfony-app/src/Entity/Teleconsultation.php
<?php

namespace App\Entity;

use App\Repository\TeleconsultationRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: TeleconsultationRepository::class)]
#[ORM\Table(name: 'teleconsultations')]
class Teleconsultation
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(name: 'initiator_id', nullable: false, onDelete: 'CASCADE')]
    private User $initiator;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(name: 'recipient_id', nullable: false, onDelete: 'CASCADE')]
    private User $recipient;

    #[ORM\Column(type: 'string', length: 255)]
    private string $roomName;

    #[ORM\Column(type: Types::TEXT, nullable: true)]
    private ?string $description = null;

    #[ORM\Column(type: Types::DATETIMETZ_IMMUTABLE)]
    private \DateTimeImmutable $scheduledAt;

    #[ORM\Column(type: Types::DATETIMETZ_IMMUTABLE, nullable: true)]
    private ?\DateTimeImmutable $startedAt = null;

    #[ORM\Column(type: Types::DATETIMETZ_IMMUTABLE, nullable: true)]
    private ?\DateTimeImmutable $endedAt = null;

    #[ORM\Column(type: 'string', length: 50, options: ['default' => 'pending'])]
    private string $status = 'pending'; // pending, ongoing, completed, cancelled

    #[ORM\Column(type: 'integer', nullable: true)]
    private ?int $durationSeconds = null;

    #[ORM\Column(type: 'string', length: 50, nullable: true)]
    private ?string $type = 'general'; // general, emergency, follow_up, etc.

    #[ORM\Column(type: Types::DATETIMETZ_IMMUTABLE)]
    private \DateTimeImmutable $createdAt;

    public function __construct()
    {
        $this->createdAt = new \DateTimeImmutable();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getInitiator(): User
    {
        return $this->initiator;
    }

    public function setInitiator(User $initiator): self
    {
        $this->initiator = $initiator;
        return $this;
    }

    public function getRecipient(): User
    {
        return $this->recipient;
    }

    public function setRecipient(User $recipient): self
    {
        $this->recipient = $recipient;
        return $this;
    }

    public function getRoomName(): string
    {
        return $this->roomName;
    }

    public function setRoomName(string $roomName): self
    {
        $this->roomName = $roomName;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): self
    {
        $this->description = $description;
        return $this;
    }

    public function getScheduledAt(): \DateTimeImmutable
    {
        return $this->scheduledAt;
    }

    public function scheduleAt(\DateTimeImmutable $scheduledAt): self
    {
        $this->scheduledAt = $scheduledAt;
        return $this;
    }

    public function getStartedAt(): ?\DateTimeImmutable
    {
        return $this->startedAt;
    }

    public function startAt(?\DateTimeImmutable $startedAt): self
    {
        $this->startedAt = $startedAt;
        return $this;
    }

    public function getEndedAt(): ?\DateTimeImmutable
    {
        return $this->endedAt;
    }

    public function endAt(?\DateTimeImmutable $endedAt): self
    {
        $this->endedAt = $endedAt;
        return $this;
    }

    public function getStatus(): string
    {
        return $this->status;
    }

    public function setStatus(string $status): self
    {
        $this->status = $status;
        return $this;
    }

    public function getDurationSeconds(): ?int
    {
        return $this->durationSeconds;
    }

    public function setDurationSeconds(?int $durationSeconds): self
    {
        $this->durationSeconds = $durationSeconds;
        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(?string $type): self
    {
        $this->type = $type;
        return $this;
    }

    public function getCreatedAt(): \DateTimeImmutable
    {
        return $this->createdAt;
    }

    public function isOngoing(): bool
    {
        return $this->status === 'ongoing';
    }

    public function isPending(): bool
    {
        return $this->status === 'pending';
    }

    public function isRequested(): bool
    {
        return $this->status === 'requested';
    }

    public function isCompleted(): bool
    {
        return $this->status === 'completed';
    }

    public function isCancel(): bool
    {
        return $this->status === 'cancelled';
    }

    /**
     * Get duration formatted
     */
    public function getDurationFormatted(): string
    {
        if (!$this->durationSeconds) {
            return '-';
        }

        $hours = intdiv($this->durationSeconds, 3600);
        $minutes = intdiv($this->durationSeconds % 3600, 60);
        $seconds = $this->durationSeconds % 60;

        $parts = [];
        if ($hours > 0) $parts[] = $hours . 'h';
        if ($minutes > 0) $parts[] = $minutes . 'm';
        if ($seconds > 0) $parts[] = $seconds . 's';

        return implode(' ', $parts) ?: '0s';
    }
}
